src/Desktop.Analyzers/Core/TypesShouldNotExtendCertainBaseTypes.cs

```diff
@@ -31,79 +34,114 @@ public abstract class TypesShouldNotExtendCertainBaseTypesAnalyzer : DiagnosticA
                                                                              s_localizableMessageSystemXmlXmlDocument,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182171.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
         internal static DiagnosticDescriptor SystemApplicationExceptionRule = new DiagnosticDescriptor(RuleId,
```

Comment:
As @mavasani suggested, these DiagnosticDescriptor all have same RuleId therefore should be combined into one. An exaample can be found here: https://github.com/mavasani/roslyn-analyzers/blob/7bc917aa3e90dcc9ff1b0081ea0187c824ceb22b/src/Microsoft.Maintainability.Analyzers/Core/DoNotIgnoreMethodResults.cs

This issue has a detailed decription for the problem: https://github.com/dotnet/roslyn-analyzers/issues/745